justfile

```diff
@@ -1,13 +1,26 @@
 default:
     @just --list
 
-build:
-    @just gateway/build
-    yarn build
-dev:
-    yarn dev
 test:
     @just gateway/test
 
+generate:
+    cd ./web-portal/backend && pnpm install && npx prisma generate
 migrate:
-    yarn migrate
+    cd ./web-portal/backend && pnpm install && npx prisma migrate dev
+
+
+dev-backend:
+    cd ./web-portal/backend && pnpm install && pnpm start
+build-backend:
+    cd ./web-portal/frontend && pnpm install && pnpm build
+serve-backend:
+    cd ./web-portal/backend && pnpm install && pnpm start:prod
+
+
+dev-frontend:
+    cd ./web-portal/frontend && pnpm install && pnpm dev
+build-frontend:
+    cd ./web-portal/frontend && pnpm install && pnpm build
+serve-frontend:
+    cd ./web-portal/frontend && pnpm install && pnpm start
```

Comment:
If you put a justfile in the subdirectory you can do `@just web-portal/build` to define the actions closer to the code and avoid the cd. This is fine though. It might be good to bundle all the builds into a `build-all` or just `build`.